theme/blog/index.html.ep

```diff
@@ -1,9 +1,22 @@
-<p>This is the blog</p>
+<h1>News</h1>
 
-<p>And these are the posts:</p>
 
-<ul>
 % for my $page ( @$pages ) {
-    <li><a href="<%= $app->page_url($page) %>"><%= $page->title %></a></li>
+%   my $d = $page->date;
+%   $d =~ s/T.*//;
+%   my @sections = $page->sections;
+<section>
+  <h2><a href="<%= $app->page_url($page) %>"><%= $page->title %></a></h2>
+  <div class="metadata">
+    <span class="date"><%= $d %><% if ($page->author) { %></span>&nbsp;-&nbsp;<span class="author"><%= $page->author %></span><% } %>
+  </div>
+
+  <%= $sections[0] %>
+
+% if ($page->sections > 1) {
```

Comment:
We already have the sections in `@sections`, so it might not be necessary to retrieve them again just to count them.

Reply:
Changed.